src/ServiceClient/Implements/DeribitSocketConnection.cs

```diff
@@ -0,0 +1,63 @@
+﻿using ServiceClient.Abstractions;
+using ServiceClient.DTOs;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ServiceClient.Implements
+{
+    public class DeribitSocketConnection : IDeribitSocketConnection
+    {
+        private readonly WebSocketOptions options;
+
+        private ClientWebSocket? socket;
+
+        public DeribitSocketConnection(WebSocketOptions options) 
+        { 
+            this.options = options;
+        }
+
+        public ClientWebSocket ClientWebSocket
+        {
+            get
+            {
+                if (socket != null)
+                    return socket;
+                socket = new ClientWebSocket
+                {
+                    Options =
+                    {
+                        KeepAliveInterval = TimeSpan.FromSeconds(options.KeepAliveIntervalInSeconds)
+                    }
+                };
+                return socket;
+            }
+        }
+
+        public async Task SocketConnectAsync(CancellationToken cancellationToken)
+        {
+            var url = new Uri(options.Url);
+
+            await ClientWebSocket
+                .ConnectAsync(url, cancellationToken)
+                .Timeout(options.ConnectionTimeoutInMilliseconds);
+        }
+
+        public async Task SocketDisconnectAsync(CancellationToken cancellationToken)
+        {
```

Comment:
the method signature could be Task<bool>... in order to know whether the socket gets disconnected.
In the implementation, I suggest adding try... catch statement

Reply:
OK

Reply:
Disagree. This a basic implementation  method and does not keeps the status of the socket, the calling method should be the one handling the exception, log it, etc.